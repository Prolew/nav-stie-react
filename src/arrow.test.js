import React from 'react';
import Arrow, {calculateArrowMarginLeft} from './arrow';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('<Arrow />', () => {
  let display = 'block';
  let fadeOut = false;
  const fromData = {height: 270, width: 420, index: 0, left: 100};
  const toData = {height: 270, width: 420, index: 0, left: 100};
  const onClick = jest.fn();
  let background = 'white';
  let leftOffset = 0;
  let rightOffset = 0;

  const props = {
    display,
    fadeOut,
    fromData,
    toData,
    onClick,
    background,
    leftOffset,
    rightOffset,
  };

  it('should fade in on cold start', () => {
    const arrow = renderer.create(<Arrow {...props}/>);
    expect(arrow).toMatchSnapshot();
  });

  it('should fade out on mouse out', () => {
    const propsCopy = {
      ...props,
      display: 'none',
      fadeOut: true,
    };
    const arrow = renderer.create(<Arrow {...propsCopy}/>);
    expect(arrow).toMatchSnapshot();
  });

  it('animate moving between content groups', () => {
    props.toData = {height: 350, width: 200, index: 1, left: 200};
    const propsCopy = {
      ...props,
      toData: {height: 350, width: 200, index: 1, left: 200},
    };
    const arrow = renderer.create(<Arrow {...propsCopy}/>);
    expect(arrow).toMatchSnapshot();
  });

  it('should compensate for left offset', () => {
    const marginLeft = calculateArrowMarginLeft(fromData, 50, 0);
    expect(marginLeft).toMatchSnapshot();
  });

  it('should compensate for right offset', () => {
    const marginLeft = calculateArrowMarginLeft(fromData, 0, 50);
    expect(marginLeft).toMatchSnapshot();
  });
});