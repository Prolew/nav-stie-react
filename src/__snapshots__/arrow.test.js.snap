// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Arrow /> animate moving between content groups 1`] = `
.c0 {
  top: -8px;
  z-index: 1;
  position: absolute;
  margin-left: 292px;
  display: block;
  width: 0;
  height: 0;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-bottom: 8px solid white;
  -webkit-animation: kLekZG 0.28s forwards ease;
  animation: kLekZG 0.28s forwards ease;
}

<div
  className="c0"
  display="block"
  onClick={[MockFunction]}
/>
`;

exports[`<Arrow /> should compensate for left offset 1`] = `
Array [
  "margin-left:",
  "242",
  "px;",
]
`;

exports[`<Arrow /> should compensate for right offset 1`] = `
Array [
  "margin-left:",
  "342",
  "px;",
]
`;

exports[`<Arrow /> should fade in on cold start 1`] = `
.c0 {
  top: -8px;
  z-index: 1;
  position: absolute;
  margin-left: 302px;
  display: block;
  width: 0;
  height: 0;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-bottom: 8px solid white;
  -webkit-animation: eMLfYp 0.25s forwards ease;
  animation: eMLfYp 0.25s forwards ease;
}

<div
  className="c0"
  display="block"
  onClick={[MockFunction]}
/>
`;

exports[`<Arrow /> should fade out on mouse out 1`] = `
.c0 {
  top: -8px;
  z-index: 1;
  position: absolute;
  margin-left: 302px;
  display: none;
  width: 0;
  height: 0;
  border-left: 8px solid transparent;
  border-right: 8px solid transparent;
  border-bottom: 8px solid white;
  -webkit-animation: hSsjZP 0.34s forwards ease;
  animation: hSsjZP 0.34s forwards ease;
}

<div
  className="c0"
  display="none"
  onClick={[MockFunction]}
/>
`;
